elasticsearch/config.go

```diff
@@ -73,19 +73,20 @@ func (h Hosts) Validate() error {
 }
 
 func connectionConfig(config *Config) (*http.Transport, []string, error) {
-	var dial, tlsDial transport.Dialer
 	var addrs []string
 	proxy, err := httpProxyURL(config)
 	if err == nil {
 		addrs, err = addresses(config)
 	}
-	if err == nil {
-		dial, tlsDial, err = dialer(config)
+	if err != nil {
+		return &http.Transport{}, nil, nil
```

Comment:
This is changing behaviour, so errors from the above functions aren't returned. Was that intentional? I suspect not, and that instead this should be:

```suggestion
		return nil, nil, err
```

(I'd also like it if we made this code a bit less clever, and change the `if err == nil` above to standard `if err != nil`, even if it does lead to duplicated code...)

Reply:
I wanted to keep the changes minimal, but happy to switch the error checks to standard behavior.